articles/user-guide/language/expressions.md

```diff
@@ -0,0 +1,558 @@
+The new expression will be the same type as the element type of the array.
+For instance, if `a` is bound to an array of `Double`s, then `a[4]` is a `Double` expression.
+
+If the array expression is not a simple identifier, it must be enclosed in parentheses in order to select an element.
+For instance, if `a` and `b` are both arrays of `Int`s, an element from the concatenation would be expressed as:
+
+```qsharp
+(a+b)[13]
+```
+
+All arrays in Q# are zero-based.
+That is, the first element of an array `a` is always `a[0]`.
+
+
+### Array Slices
+
+Given an array expression and a `Range` expression, a new expression may be formed using the `[` and `]` array slice operator.
+The new expression will be the same type as the array and will contain the array items indexed by the elements of the `Range`, in the order defined by the `Range`.
+For instance, if `a` is bound to an array of `Double`s, then `a[3..-1..0]` is a `Double[]` expression that contains the first four elements of `a` but in the reverse order as they appear in `a`.
+
+If the `Range` is empty, then the resulting array slice will be zero length.
+
+Just as with referencing array elements, if the array expression is not a simple identifier, it must be enclosed it parentheses in order to slice.
+If `a` and `b` are both arrays of `Int`s, a slice from the concatenation would be expressed as:
+
+```qsharp
+(a+b)[1..2..7]
+```
+
+#### Inferred start/end values
+
+Starting with our 0.8 release, we support contextual expressions for range slicing. 
+In particular, range start and end values may be omitted in the context of a range slicing expression. 
+In that case, the compiler will apply the following rules to infer the intended delimiters for the range. 
+
+For example, if the range start value is omitted,  then the inferred start value 
+- is zero if no step is specified or the specified step is positive, and 
+- is the length of sliced array minus one if the specified step is negative. 
+
+If the range end value is omitted,  then the inferred end value 
+- is the length of sliced array minus one if no step is specified or the specified step is positive, and 
+- is zero if the specified step is negative. 
+
+```qsharp
+let arr = [1,2,3,4,5,6];
+let slice1  = arr[3...];      // slice1 is [4,5,6];
+let slice2  = arr[0..2...];   // slice2 is [1,3,5];
+let slice3  = arr[...2];      // slice3 is [1,2,3];
+let slice4  = arr[...2..3];   // slice4 is [1,3];
+let slice5  = arr[...2...];   // slice5 is [1,3,5];
+let slice7  = arr[4..-2...];  // slice7 is [5,3,1];
+let slice8  = arr[...-1..3];  // slice8 is [6,5,4];
+let slice9  = arr[...-1...];  // slice9 is [6,5,4,3,2,1];
+let slice10 = arr[...];       // slice10 is [1,2,3,4,5,6];
+```
+
+### Copy-and-Update Expressions
+
+Since all Q# types are value types---with the qubits taking a somewhat special role---formally a "copy" is created when a value is bound to a symbol, or when a symbol is rebound. 
```

Comment:
```suggestion
Since all Q# types are value types — with the qubits taking a somewhat special role —formally a "copy" is created when a value is bound to a symbol, or when a symbol is rebound. 
```